### ISSUE TITLE
Allow Overwrite of Existing Variables

### ISSUE THREAD

**Issue by sheldonhull** [terraform-providers/terraform-provider-tfe#105]
## Current Problem Of Existing Variables
If I want to update all my workspaces with `tfe_variable`, I find that it errors instead of replacing if the variable already exists. 

I know this matters for typically terraform server behavior. However, having an option to overwrite if exists would be great, as then existing workspaces can be easily managed by updating the variables and ensuring terraform starts managing. Right now, I have to go into the workspace in Terraform Cloud, and manually delete each variable, and then then rerun the plan to reapply all the missing variables. 

This results in a very brittle process if someone adds a variable manually due to time constraints, as the variable creation will then fail. 

## Possible Solution

Add parameter for `allow_overwrite` or `force` that would ensure if the variable already exists it would be deleted and then the apply of the variable would be allowed to proceed without error.

**Comment by andres-de-castro** [comment]
I am not seeing this problem

**Comment by sheldonhull** [comment]
to reproduce...
- use new workspace
- create variable in web gui
- now try to create this variable via terraform

it should fail
I'd expect it to instead just overwrite the variable without error

**Comment by svanharmelen** [comment]
While I understand that it can be a pain to import all variables one by one, I don't agree that it should overwrite the variables without error. That kind of (IMHO) defeats the guarantees Terraform tries to offer and doesn't match the model that is used by Terraform internally (as each variable is an independent resource from the API's point of view).

**Comment by sheldonhull** [comment]
Is there any lifecycle option that could clear the existing variable if exists so it's a recreate regardless if the variable exists or not? 

I totally agree with your perspective on the model terraform uses. However, as this is a workspace configuration, trying to figure out how to avoid more toil on just managing these variables if they exist without having to click in the gui 8 times. 😁

**Comment by svanharmelen** [comment]
> Is there any lifecycle option that could clear the existing variable if exists so it's a recreate regardless if the variable exists or not?

No, there isn't... Maybe it would be a good idea to write a small script or little Go program that uses the API to delete all the variables just before you run Terraform on the workspace for the first time. This could be done by using these API calls: https://www.terraform.io/docs/enterprise/api/variables.html (implemented in Go [here](https://github.com/hashicorp/go-tfe/blob/master/variable.go).

Additionally you can also choose (maybe even a better option) to import the existing variables to bring them under Terraform control: https://www.terraform.io/docs/providers/tfe/r/variable.html#import

**Comment by barrettclark** [comment]
Closing due to inactivity
